packages/app/src/cli/models/extensions/specifications/transform/app_config_webhook.ts

```diff
@@ -24,12 +24,96 @@ export function transformToWebhookConfig(content: object) {
+    .filter((subscription) => subscription.topics !== undefined)
+    .map((subscription) => {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      const {compliance_topics, ...rest} = subscription
+      return {topics: subscription.topics, ...rest}
+    })
+  const complianceSubscriptions = subscriptions
+    .filter((subscription) => subscription.topics === undefined || subscription.compliance_topics !== undefined)
+    .map((subscription) => {
+      const {topics, ...rest} = subscription
+      return {compliance_topics: subscription.compliance_topics, ...rest}
+    })
+
+  const mergedTopicSubscriptions = reduceWebhooks(topicSubscriptions, 'topics')
+  const mergedComplianceSubscriptions = reduceWebhooks(complianceSubscriptions, 'compliance_topics')
+  const sortedTopicsSubscriptions = sortTopics(mergedTopicSubscriptions)
+  const sortedComplianceTopicsSubscriptions = sortComplianceTopics(mergedComplianceSubscriptions)
+
+  return [...sortWebhooksByUri(sortedTopicsSubscriptions), ...sortWebhooksByUri(sortedComplianceTopicsSubscriptions)]
+}
+
+function sortArrayAlphabetically(array: string[] | undefined) {
+  return array?.sort((first, second) => first.localeCompare(second))
+}
+function sortWebhooksByUri(subscriptions: WebhookSubscription[]) {
+  return subscriptions.sort((oneSub, twoSub) => oneSub.uri.localeCompare(twoSub.uri))
+}
+function sortTopics(subscriptions: WebhookSubscription[]) {
+  const topicsSortedSubscriptions = subscriptions.map((subscription) => {
+    const sortedTopics = sortArrayAlphabetically(subscription.topics)
+    const {topics, ...rest} = subscription
+    return {topics: sortedTopics, ...rest}
+  })
+  return topicsSortedSubscriptions
+}
+function sortComplianceTopics(subscriptions: WebhookSubscription[]) {
+  const topicsSortedSubscriptions = subscriptions.map((subscription) => {
+    const sortedTopics = sortArrayAlphabetically(subscription.compliance_topics)
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const {compliance_topics, ...rest} = subscription
+    return {compliance_topics: sortedTopics, ...rest}
+  })
+  return topicsSortedSubscriptions
+}
+
+function findSubscription(subscriptions: WebhookSubscription[], subscription: WebhookSubscription) {
+  return subscriptions.find(
+    (sub) =>
+      sub.uri === subscription.uri &&
+      sub.sub_topic === subscription.sub_topic &&
+      deepCompare(sub.include_fields ?? [], subscription.include_fields ?? []) &&
+      sub.filter === subscription.filter,
+  )
+}
+
+export function reduceWebhooks(
+  subscriptions: WebhookSubscription[],
+  property?: keyof Pick<WebhookSubscription, 'topics' | 'compliance_topics'>,
+) {
+  return subscriptions.reduce((accumulator, subscription) => {
+    const existingSubscription = findSubscription(accumulator, subscription)
+    if (existingSubscription) {
+      if (property) {
+        ;(existingSubscription[property] ?? []).push(...(subscription[property] ?? []))
```

Comment:
this is odd... syntax and all. what would you be `push`ing into an empty array for?

```suggestion
      if (property && existingSubscription[property].length) {
        ;(existingSubscription[property] ?? []).push(...(subscription[property] ?? []))
```

should help... can we also use [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) here or is the spread supposed to be some special overwriting under the hood?

Reply:
Not sure, I think @isaacroldan might have a better idea about this? I think the spread does have special overwriting under the hood, cause `concat` breaks a couple of tests I had for this

Also unfortunately `existingSubscription[property].length` doesn't help with needing the empty array as this can also be `undefined` :/